server/src/main/java/org/apache/iotdb/db/rescon/SystemInfo.java

```diff
@@ -170,6 +176,27 @@ public synchronized void resetFlushingMemTableCost(long flushingMemTableCost) {
     this.flushingMemTablesCost -= flushingMemTableCost;
   }
 
+  public void addCompactionMemoryCost(long memoryCost) throws InterruptedException {
+    long originSize = this.compactionMemoryCost.get();
+    while (originSize + memoryCost > memorySizeForCompaction
```

Comment:
If there are two tasks call this method concurrently, 
Task1 originSize = 10;
Task2 originSize = 10;
Task1.compactionMemoryCost.compacreAndSet(10, 10+30) = 40;
Task2 compareAndSet is always false, and Task2 sleeps forever.

Reply:
The originSize is updated in the while block.